Controller/RoleController.php

```diff
@@ -116,8 +117,15 @@ public function createRoleAction(Request $request)
      */
     public function updateRoleAction(Request $request, $roleId)
     {
-        $role = $this->roleService->loadRole($roleId);
-        $roleData = (new RoleMapper())->mapToFormData($role);
+        try {
+            $roleDraft = $this->roleService->loadRoleDraft($roleId);
+        } catch (NotFoundException $e) {
+            // The draft doesn't exist, let's create one
+            $role = $this->roleService->loadRole($roleId);
+            $roleDraft = $this->roleService->createRoleDraft($role);
+        }
+
+        $roleData = (new RoleMapper())->mapToFormData($roleDraft);
```

Comment:
Given what those mappers do, this would really be overkill. Same as for form types, when they are simple (they usually are), it's perfectly fine to instantiate them instead of defining services. Besides there is absolutely no dependency.
Only thing is that it's not extensible, but is it really needed here?